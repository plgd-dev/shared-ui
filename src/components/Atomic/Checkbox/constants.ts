export const checkboxValues = {
    ON: 'on',
    OFF: 'off',
} as const
