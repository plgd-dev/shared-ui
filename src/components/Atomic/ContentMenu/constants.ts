export const statuses = {
    SUCCESS: 'success',
    ERROR: 'error',
    WARNING: 'warning',
} as const
