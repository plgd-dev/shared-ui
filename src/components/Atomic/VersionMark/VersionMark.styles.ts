import { css } from '@emotion/react'
import { colors } from '../_utils/colors'
import { ThemeType, getThemeColor, getTheme } from '../_theme'
import { fontPrimary } from '../_utils/commonStyles'

export const versionMark = css`
    display: flex;
    align-items: center;
`

export const left = css`
    flex: 0 0 24px;
`

export const right = css`
    flex: 1 1 auto;
    padding-left: 12px;
`

export const versionText = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    text-align: center;
    color: ${colors.neutral500};
`

export const updateText = (theme: ThemeType) => css`
    color: ${getThemeColor(theme, `VersionMark.link.color`)};
    text-decoration: underline;
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-style: normal;
    font-weight: 300;
    font-size: 12px;
    line-height: 18px;
    padding-left: 6px;
    cursor: pointer;

    &:hover {
        text-decoration: none;
    }
`
