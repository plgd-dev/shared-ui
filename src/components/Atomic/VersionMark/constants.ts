export const severities = {
    SUCCESS: 'success',
    WARNING: 'warning',
    ERROR: 'error',
} as const
