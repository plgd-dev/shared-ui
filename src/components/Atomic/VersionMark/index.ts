export { default } from './VersionMark'
export * from './VersionMark'

export * from './utils'
