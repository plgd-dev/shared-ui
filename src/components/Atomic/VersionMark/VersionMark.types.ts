import { SyntheticEvent } from 'react'
import { severities } from './constants'

export type VersionMarkSeverityType = (typeof severities)[keyof typeof severities]

export type Props = {
    className?: string
    collapsed?: boolean
    id?: string
    update?: {
        text: string
        onClick?: (e: SyntheticEvent) => void
    }
    versionText: string
    severity: VersionMarkSeverityType
}

export const defaultProps = {
    collapsed: false,
}
