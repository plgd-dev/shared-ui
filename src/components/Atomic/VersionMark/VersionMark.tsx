import { FC } from 'react'
import { defaultProps, Props, VersionMarkSeverityType } from './VersionMark.types'
import * as styles from './VersionMark.styles'
import { severities } from './constants'
import { convertSize, IconCloudError, IconCloudSuccess, IconCloudWarning } from '../Icon'

const VersionMark: FC<Props> = (props) => {
    const { collapsed, severity, update, versionText, className, id } = { ...defaultProps, ...props }

    const getIcon = (severity: VersionMarkSeverityType) => {
        switch (severity) {
            case severities.SUCCESS:
                return <IconCloudSuccess {...convertSize(24)} />
            case severities.WARNING:
                return <IconCloudWarning {...convertSize(24)} />
            case severities.ERROR:
                return <IconCloudError {...convertSize(24)} />
        }
    }

    return (
        <div className={className} css={styles.versionMark} id={id}>
            <div css={styles.left}>{getIcon(severity)}</div>
            {!collapsed && (
                <div css={styles.right}>
                    <span css={styles.versionText}>{versionText}</span>
                    {update && (
                        <span css={styles.updateText} onClick={update.onClick}>
                            {update.text}
                        </span>
                    )}
                </div>
            )}
        </div>
    )
}

VersionMark.displayName = 'VersionMark'
VersionMark.defaultProps = defaultProps

export default VersionMark
