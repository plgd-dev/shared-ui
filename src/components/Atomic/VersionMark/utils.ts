import { VersionMarkSeverityType } from './VersionMark.types'
import { severities } from './constants'

export type Props = {
    buildVersion: string
    githubVersion: string
    i18n: {
        newUpdateIsAvailable: string
        version: string
    }
}

export type getVersionMarkDataReturnType = {
    severity: VersionMarkSeverityType
    text: string
}

export const getVersionMarkData = (props: Props): getVersionMarkDataReturnType => {
    const { buildVersion, githubVersion, i18n } = props
    let text = `${i18n.version} ${buildVersion}`

    if (githubVersion === buildVersion) {
        return {
            severity: severities.SUCCESS,
            text,
        }
    } else {
        text += ` â€¢ ${i18n.newUpdateIsAvailable}`
        const [latestMajor, latestMinor, latestPatch] = githubVersion?.split('-') || []
        const [currentMajor, currentMinor, currentPath] = buildVersion.split('-')[0].split('.')
        const hasError =
            Math.abs(parseInt(latestMinor) - parseInt(currentMinor)) >= 3 ||
            Math.abs(parseInt(latestPatch) - parseInt(currentPath)) >= 3 ||
            latestMajor !== currentMajor

        return {
            severity: hasError ? severities.ERROR : severities.WARNING,
            text,
        }
    }
}
