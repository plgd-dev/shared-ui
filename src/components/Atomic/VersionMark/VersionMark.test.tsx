import React from 'react'
import { render } from '@testing-library/react'
import VersionMark from './VersionMark'
import { severities } from './constants'

describe('<VersionMark>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <VersionMark severity={severities.SUCCESS} versionText='Version 2.02' />
                <VersionMark
                    severity={severities.WARNING}
                    update={{
                        text: 'Click here!',
                        onClick: () => console.log('update click'),
                    }}
                    versionText='Version 2.02 • New update is available.'
                />
                <VersionMark
                    severity={severities.ERROR}
                    update={{
                        text: 'Click here!',
                        onClick: () => console.log('update click'),
                    }}
                    versionText='Version 2.02 • New update is available.'
                />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
