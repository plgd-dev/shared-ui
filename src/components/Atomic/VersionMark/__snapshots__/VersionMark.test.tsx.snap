// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VersionMark> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  -webkit-flex: 0 0 24px;
  -ms-flex: 0 0 24px;
  flex: 0 0 24px;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding-left: 12px;
}

.emotion-3 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 18px;
  text-align: center;
  color: #81868C;
}

.emotion-8 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 300;
  font-size: 12px;
  line-height: 18px;
  padding-left: 6px;
  cursor: pointer;
}

.emotion-8:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M24 9.651c0 1.45-.465 2.785-1.246 3.853v.001c-.954 1.275-2.02 1.787-2.287 1.915l-.037.018c-1.16.56-2.382.582-2.95.554l-.048.001h-.058a678 678 0 0 0-6.006 0c-.875 0-1.94-.005-3.022-.01-1.116-.007-2.25-.013-3.209-.013C2.3 15.97 0 13.56 0 10.586c0-2.973 2.3-5.384 5.137-5.384l.096.002.048.002.001.007.002.006C5.953 2.228 8.514 0 11.57 0c2.064 0 3.9 1.014 5.083 2.592.105.14.276.41.492.773q.394-.057.801-.058C21.29 3.307 24 6.147 24 9.651m-7.765-3.872a.99.99 0 0 1 0 1.349l-4.798 5.092a.86.86 0 0 1-1.272 0L7.763 9.674a.996.996 0 0 1 0-1.349.86.86 0 0 1 1.272 0l1.764 1.87 4.165-4.416a.864.864 0 0 1 1.271 0"
            fill="#52C5A2"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="emotion-2"
      >
        <span
          class="emotion-3"
        >
          Version 2.02
        </span>
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M24 9.651c0 1.45-.465 2.785-1.246 3.853v.001c-.954 1.275-2.02 1.787-2.287 1.915l-.037.018c-1.16.56-2.382.582-2.95.554l-.048.001h-.058a678 678 0 0 0-6.006 0c-.875 0-1.94-.005-3.022-.01-1.116-.007-2.25-.013-3.209-.013C2.3 15.97 0 13.56 0 10.586c0-2.973 2.3-5.384 5.137-5.384l.096.002.048.002.001.007.002.006C5.953 2.228 8.514 0 11.57 0c2.064 0 3.9 1.014 5.083 2.592.105.14.276.41.492.773q.394-.057.801-.058C21.29 3.307 24 6.147 24 9.651M11.291 4.264a1.08 1.08 0 0 1 1.415 0h.003l3.998 3.6a.84.84 0 0 1 0 1.274 1.08 1.08 0 0 1-1.415 0L13 7.074V12.1c0 .498-.446.9-.999.9s-1-.402-1-.9V7.074L8.709 9.138a1.08 1.08 0 0 1-1.415 0 .84.84 0 0 1 0-1.274z"
            fill="#EAAA43"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="emotion-2"
      >
        <span
          class="emotion-3"
        >
          Version 2.02 • New update is available.
        </span>
        <span
          class="emotion-8"
        >
          Click here!
        </span>
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M24 9.651c0 1.45-.465 2.785-1.246 3.853v.001c-.954 1.275-2.02 1.787-2.287 1.915l-.037.018c-1.16.56-2.382.582-2.95.554l-.048.001h-.058a678 678 0 0 0-6.006 0c-.875 0-1.94-.005-3.022-.01-1.116-.007-2.25-.013-3.209-.013C2.3 15.97 0 13.56 0 10.586c0-2.973 2.3-5.384 5.137-5.384l.096.002.048.002.001.007.002.006C5.953 2.228 8.514 0 11.57 0c2.064 0 3.9 1.014 5.083 2.592.105.14.276.41.492.773q.394-.057.801-.058C21.29 3.307 24 6.147 24 9.651M11 5a1 1 0 1 1 2 0v4a1 1 0 1 1-2 0zm0 7a1 1 0 1 1 2 0 1 1 0 0 1-2 0"
            fill="#D74E3A"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="emotion-2"
      >
        <span
          class="emotion-3"
        >
          Version 2.02 • New update is available.
        </span>
        <span
          class="emotion-8"
        >
          Click here!
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
